enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,423 @@
+dropped.
+- the metrics in the `keep` section were obtained with the help of a script that
+  parsed all Alerts, PrometheusRules and Console dashboards to determine what
+  metrics were actually being used.
+
+Finally, a team that adopts the metrics collection profile feature should also 
+add themselves and their component to the list under 
+[Infrastruture needed](#infrastructure-needed-optional)
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+```go
+type PrometheusK8sConfig struct {
+	// Defines the metrics collection profile that Prometheus uses to collect
+	// metrics from the platform components. Supported values are `full` or
+	// `minimal`. In the `full` profile (default), Prometheus collects all
+	// metrics that are exposed by the platform components. In the `minimal`
+	// profile, Prometheus only collects metrics necessary for the default
+	// platform alerts, recording rules, telemetry and console dashboards.
+	CollectionProfile CollectionProfile `json:"collectionProfile,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+Each OpenShift team that wants to adopt this feature will be responsible for
+providing the different monitors. This work is not trivial. Dependencies between
+operators and their metrics exist. This makes it difficult for developers to
+determine whether a given metric can be excluded from the `minimal` profile or
+not. To aid teams with this effort the monitoring team will provide:
+- a CLI tool that consumes a ServiceMonitor and a list of metrics and
+  generates the `minimal` ServiceMonitor for that operator.
+- an origin/CI test that validates for all Alerts and PrometheusRules that the
+  metrics used by them are present in the `keep` expression of the
+  ServiceMonitor for the `minimal` profile
+
+
+### Risks and Mitigations
+
+- How are monitors supposed to be kept up to date? In 4.12 a metric that wasn't
+  being used in an alert is now required, how does the monitor responsible for
+  that metric gets updated?
+  - The origin/CI test mentioned in the previous section will fail if ther is a
+    resouce (Alerts/PrometheusRules/Dashboards) using a metric which is not
+    present in the ServiceMonitor;
+
+- A new profile is added, what will happen to operators that had implemented
+  metrics collection profiles but did not implement the latest profile.
+  - The monitoring team will use the list under 
+  [Infrastruture needed](#infrastructure-needed-optional) to help the developers
+  with the addoption of the new metrics collection profile.
+  - Furthermore, we plan on adding extra origin/CI tests to validate each new
+    profile if applicable.
+
+- What happens if CMO pick-up an unsupported collection profile value?
+  - Report Degraded=False and fail reconciliation.
+
+### Drawbacks
+
+- Extra CI cycles
+
+## Design Details
+
+### Open Questions
+
+- Should we add future profiles? 
```

Comment:
I think another open question would be, if we'd add another profile, how would we validate it?
Our current validation strategy consists of comparing metrics we use (In Rules and such) against regexes in relabel keep actions. 
For hypothetical future profiles this strategy is unclear. Since the process of adding a new profile is also a little hand-wavy (keep a list of teams to follow up with) maybe it could be an option to declare adding another profile outside of this particular proposal?
This feature is valuable as is (with only the full and minimal profile) and its unclear whether we'll add a third, so maybe we should exclude this scenario explicitly?

Reply:
What if we move this to `Risks and Mitigations` and detail some of the topics you highlighted there? I'm also happy with just moving it to a `Non goal` but I believe having it detailed in `Risks and Mitigations` might bring more value in the long run. WDYT?

Reply:
I did as I suggested in last commit, if you feel I didn't hit the mark I'm happy with moving it to the `Non goals`

Reply:
(+1 on moving profile additions outside of this proposal) 🙂